filters.go

```diff
@@ -311,6 +311,38 @@ func (p *Pipe) MatchRegexp(re *regexp.Regexp) *Pipe {
 	})
 }
 
+// ReadInput reads displays the pipe content as a message as a prompt
+// and reads the content from standard input and returns the result
+// as a pipe. In case the input was empty it defaults to its `defaultValue`
+func (p *Pipe) ReadInput(defaultValue string) *Pipe {
+	const emptyString = ""
+	const withDefault = "%s [%s]: "
+	const withoutDefault = "%s: "
+	if p == nil || p.Error() != nil {
+		return p
+	}
+	defer p.Close()
+
+	prompt, _ := p.String() // if pipe is empty, prompt message is empty
+
+	if defaultValue == emptyString {
+		Echo(fmt.Sprintf(withoutDefault, prompt)).Stdout()
+	} else {
+		Echo(fmt.Sprintf(withDefault, prompt, defaultValue)).Stdout()
+	}
+
+	scanner := bufio.NewScanner(Stdin().Reader)
```

Comment:
Did you think about something like `fmt.Scanln` instead here?

Reply:
I considered and tried it out, but ultimately decided against it. 

First, `fmt.Scanln` and the entire `fmt.Scan*` family seem to scan space-separated tokens. Plus, `bufio.NewScanner(io.Reader)` is easier to stub and be abstracted.